### PR TITLE
Feature/154 check compatibility with android project with active pitest config

### PR DESCRIPTION
**Pull request by nikomall34**
I didn't have to change a lot in the code. I updated gradle to a higher version and the gradle build system has different life-cycles. So right now we override our pitest settings at the correct life-cycle.
In case the android project is not running on your machine you can try the following steps:
1. File/Repair IDE (you can do this for the sandbox-ide and the one you have installed locally)
2. File/Invalidate Caches (you can do this for the sandbox-ide and the one you have installed locally)
3. Delete all build, .gradle and .idea folders. (Also in the projects that you open with the sandbox-ide)
4. Check for updates in your IDEA and install them
5. Install the Android plugin in the sandbox-ide
6. Contact other SDs in Discord or here

If you never worked with the android-project @emuguy1 adjusted it slightly [here](https://github.com/Pfoerd/pitest-kotlin-android-sample/tree/add_pitest_support)

I configured a run configuration for the `runIde` task by introducing a new task named `androidRunIde`. Given that the installation path of Android Studio may vary among us, I introduced a property in the `gradle.properties` file named `androidStudioHome`. Each developer is required to set this property locally, which is why I included the gradle.properties file in the .gitignore. In case you don't set the property the `androidRunIde` will just start the IntelliJ IDEA.

### CODE REVIEW

File: pitmutationmate/build.gradle.kts [hunk 1]
```diff
@@ -65,4 +65,11 @@ tasks {
     publishPlugin {
         token.set(System.getenv("PUBLISH_TOKEN"))
     }
+
+    register<org.jetbrains.intellij.tasks.RunIdeTask>("androidRunIde") {
```
Comment (lheimbs) [inline pitmutationmate/build.gradle.kts:69]: I think `runIdeAndroid` is more fitting and in line with other task names...
Or even `runIdeAndroidStudio`

File: pitmutationmate/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-bin.zip
```
Comment (lheimbs) [inline pitmutationmate/gradle/wrapper/gradle-wrapper.properties:6]: why not up to 8.5 while were at it?

File: coverage-plugin/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -4,6 +4,6 @@
 #Fri Dec 22 14:41:56 CET 2023
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
```
Comment (lheimbs) [inline coverage-plugin/gradle/wrapper/gradle-wrapper.properties:7]:
```suggestion
```
I think artifact from other PR

File: pitmutationmate/gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (nikomall34) [reply pitmutationmate/gradle/wrapper/gradle-wrapper.properties:6]: The Problem is that when I used gradle 8.5 I got the the following error during the github jobs:
```
Exception is:
java.lang.NoClassDefFoundError: org/gradle/api/publish/ivy/internal/publication/DefaultIvyPublicationIdentity
```

File: coverage-plugin/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (nikomall34) [reply coverage-plugin/gradle/wrapper/gradle-wrapper.properties:7]: Yes I fixed this on another PR. Sorry for confusing. Gradle 8.5 was working locally but than the github build and test jobs failed with the error I mentioned above.

**Review by lheimbs — APPROVED** [review]

**Pull request merged**